loki-build-image/Dockerfile

```diff
@@ -1,28 +1,35 @@
-FROM golang:1.11.4-stretch
-RUN apt-get update && apt-get install -y file jq unzip protobuf-compiler libprotobuf-dev libsystemd-dev && \
-	rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
-ENV DOCKER_VER="17.03.0-ce"
-RUN curl -L -o /tmp/docker-$DOCKER_VER.tgz https://download.docker.com/linux/static/stable/x86_64/docker-$DOCKER_VER.tgz && \
-	tar -xz -C /tmp -f /tmp/docker-$DOCKER_VER.tgz && \
-	mv /tmp/docker/* /usr/bin && \
-	rm /tmp/docker-$DOCKER_VER.tgz
-ENV HELM_VER="v2.13.1"
-RUN curl -L -o /tmp/helm-$HELM_VER.tgz http://storage.googleapis.com/kubernetes-helm/helm-${HELM_VER}-linux-amd64.tar.gz && \
-	tar -xz -C /tmp -f /tmp/helm-$HELM_VER.tgz && \
-	mv /tmp/linux-amd64/helm /usr/bin/helm && \
-	rm -rf /tmp/linux-amd64 /tmp/helm-$HELM_VER.tgz
+FROM alpine as helm
+ARG HELM_VER="v2.13.1"
+RUN apk add --no-cache curl && \
+    curl -L -o /tmp/helm-$HELM_VER.tgz http://storage.googleapis.com/kubernetes-helm/helm-${HELM_VER}-linux-amd64.tar.gz && \
+    tar -xz -C /tmp -f /tmp/helm-$HELM_VER.tgz && \
+    mv /tmp/linux-amd64/helm /usr/bin/helm && \
+    rm -rf /tmp/linux-amd64 /tmp/helm-$HELM_VER.tgz
+
+FROM alpine as golangci
+RUN apk add --no-cache curl && \
+    cd / && \
+    curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s v1.17.1
+
+FROM golang:1.11.4-alpine
+RUN sed -i -e 's/v[[:digit:]]\.[[:digit:]]/edge/g' /etc/apk/repositories && \
+    apk update && apk upgrade &&\
+    apk add --no-cache \
+      coreutils file bash unzip\
+      git make grep jq \
+      protobuf protobuf-dev \
+      docker-cli
+COPY --from=helm /usr/bin/helm /usr/bin/helm
+COPY --from=golangci /bin/golangci-lint /usr/local/bin
 RUN go get \
-	github.com/golang/protobuf/protoc-gen-go \
-	github.com/gogo/protobuf/protoc-gen-gogoslick \
-	github.com/gogo/protobuf/gogoproto \
-	github.com/go-delve/delve/cmd/dlv \
-	golang.org/x/tools/cmd/goyacc && \
-	rm -rf /go/pkg /go/src
-ENV GOLANGCI_LINT_COMMIT="692dacb773b703162c091c2d8c59f9cd2d6801db"
```

Comment:
I know we pinned this for a reason @cyriltovena maybe the release you called out above is newer and we don't need to pin to a commit hash?